utils/google.js

```diff
@@ -0,0 +1,130 @@
+import badger from './nearbadger.js';
+
+const GOOGLE_API_URL = 'https://oauth2.googleapis.com';
+const GOOGLE_OAUTH_TOKEN_ENDPOINT = 'token';
+const GOOGLE_OAUTH_USERS_INFO_ENDPOINT =  'userinfo';
+
+const GOOGLE_AUTH_URL = 'https://www.googleapis.com/oauth2/v3';
+
+export class GoogleAPI {
+    static async request(endpoint, options) {
+        return fetch(endpoint, options)
+        .then(
+            (data) => {
+                return data.json();
+            }
+        );
+    }
+    static async getUserAccessToken(params) {
+        const options = {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded',
+            },
+            body: new URLSearchParams(
+                this.getAccessTokenParams(params)
+            ),
+        };
+
+        const fullUrl = `${GOOGLE_API_URL}/${GOOGLE_OAUTH_TOKEN_ENDPOINT}`;
+
+        return this.request(fullUrl, options)
+        .then(
+            (info) => {
+                return info?.access_token || '';
+            }
+        );
+    }
+    static async getUserHandle(accessToken) {
+        const options = {
+            headers: {
+                Authorization: `Bearer ${accessToken}`
+            }
+        };
+
+        const fullUrl = `${GOOGLE_AUTH_URL}/${GOOGLE_OAUTH_USERS_INFO_ENDPOINT}`;
+
+        return this.request(fullUrl, options)
+        .then(
+            (user) => {
+                return user?.email || ''
+            }
+        );
+    }
+    static getAccessTokenParams({
+        code,
+        redirectUri,
+        codeVerifier
+    }) {
+        return {
+            code,
+            client_id: process.env.GOOGLE_OAUTH_CLIENT_ID,
```

Comment:
Please, add these two environment variables as empty strings to the `.env.example` file, so we don't lose track of the env variables.